RedfishEventListener_v1.py

```diff
@@ -309,33 +152,134 @@ def process_data(newsocketconn, fromaddr):
+    config['keyfile'] = parsed_config.get('CertificateDetails', 'keyfile')
+
+    if parsed_config.has_section("SubsciptionDetails") and parsed_config.has_section("SubscriptionDetails"):
+        my_logger.error('Use either SubsciptionDetails or SubscriptionDetails in config, not both.')
+        sys.exit(1)
+    my_config_key = "SubsciptionDetails" if parsed_config.has_section("SubsciptionDetails") else "SubscriptionDetails"
+    config['destination'] = parsed_config.get(my_config_key, 'Destination')
+    config['contextdetail'] = parsed_config.get(my_config_key, 'Context')
+    config['protocol'] = parsed_config.get(my_config_key, 'Protocol')
+    config['subscriptionURI'] = parsed_config.get(my_config_key, 'SubscriptionURI')
+    config['eventtypes'] = parse_list(parsed_config.get(my_config_key, 'EventTypes'))
+
+    config['serverIPs'] = parse_list(parsed_config.get('ServerInformation', 'ServerIPs'))
+    config['usernames'] = parse_list(parsed_config.get('ServerInformation', 'UserNames'))
+    config['passwords'] = parse_list(parsed_config.get('ServerInformation', 'Passwords'))
+    config['logintype'] = ['Session' for x in config['serverIPs']]
+    if parsed_config.has_option('ServerInformation', 'LoginType'):
+        config['logintype'] = parse_list(parsed_config.get('ServerInformation', 'LoginType'))
+        config['logintype'] += ['Session'] * (len(config['serverIPs']) - len(config['logintype']))
+
+    config['certcheck'] = parsed_config.getboolean('ServerInformation', 'certcheck')
+    config['verbose'] = args.verbose
+
+    ### Perform the Subscription if provided
+    SubscriptionURI, Protocol, ContextDetail, EventTypes, Destination = config['subscriptionURI'], config['protocol'], config['contextdetail'], config['eventtypes'], config['destination']
+
+    target_contexts = []
+
+    if not (len(config['serverIPs']) == len(config['usernames']) == len(config['passwords'])):
+        my_logger.error("Number of ServerIPs does not match UserNames, Passwords or LoginTypes")
+        sys.exit(1)
+    elif len(config['serverIPs']) == 0:
+        my_logger.info("No subscriptions are specified. Continuing with Listener.")
+    else:
+        for dest, user, passwd, logintype in zip(config['serverIPs'], config['usernames'], config['passwords'], config['logintype']):
+            try:
+                ### Create Subsciption on the servers provided by users if any
+                my_logger.info("ServerIP:: {}".format(dest))
+                my_logger.info("UserName:: {}".format(user))
+                my_ctx = redfish_client(dest, user, passwd, timeout=30)
+                my_ctx.login(auth={
+                    "Basic": AuthMethod.BASIC,
+                    "Session": AuthMethod.SESSION,
+                    "None": None
+                }[logintype])
+                response = event_service.create_event_subscription(my_ctx, config['destination'], client_context=config['contextdetail'], event_types=config['eventtypes'])
```

Comment:
There are some new config options we'd like to pass in here too:
* `format`
* `expand`
* `resource_types`
* `registries`

Reply:
Noted.  I think this is what I confused with "new/deprecated event payload keys"